spec/lib/data_hygiene/delete_topics_without_subscribers_spec.rb

```diff
@@ -0,0 +1,127 @@
+require 'rails_helper'
+require 'data_hygiene/delete_topics_without_subscribers'
+
+RSpec.describe DataHygiene::DeleteTopicsWithoutSubscribers do
+  let!(:subscriber_list) { create(:subscriber_list, title: 'Test') }
+  let(:client) { double('GovDelivery::Client') }
+  let(:input) { StringIO.new }
+  let(:output) { StringIO.new }
+  subject { described_class.new(client: client, input: input, output: output) }
+
+  describe '#with_zero_subscribers' do
+    before do
+      allow(client).to receive(:fetch_topic).and_return(fetch_topics_response)
+    end
+
+    context 'when the topic has no subscribers' do
+      let(:fetch_topics_response) { { 'subscribers_count' => 0 } }
+
+      it 'is included in the list' do
+        expect(subject.with_zero_subscribers).to include(subscriber_list)
+      end
+    end
+
+    context 'when the topic has subscribers' do
+      let(:fetch_topics_response) { { 'subscribers_count' => 2 } }
+
+      it 'is not included in the list' do
+        expect(subject.with_zero_subscribers).not_to include(subscriber_list)
+      end
+    end
+
+    context 'when the topic does not exist' do
+      let(:fetch_topics_response) { nil }
+
+      it 'is not included in the list' do
+        expect(subject.with_zero_subscribers).not_to include(subscriber_list)
+      end
+    end
+  end
+
+  describe '#with_gov_delivery_error' do
+    let!(:subscriber_list) { create(:subscriber_list) }
+    let(:client) { double('GovDelivery::Client') }
+
+    before do
+      allow(client).to receive(:fetch_topic).and_return(fetch_topics_response)
+    end
+
+    subject { described_class.new(client: client, input: input, output: output) }
+
+    context 'when the topic has no subscribers' do
+      let(:fetch_topics_response) { { 'subscribers_count' => 0 } }
+
+      it 'is not included in the list' do
+        expect(subject.with_gov_delivery_error).not_to include(subscriber_list)
+      end
+    end
+
+    context 'when the topic has subscribers' do
+      let(:fetch_topics_response) { { 'subscribers_count' => 2 } }
+
+      it 'is not included in the list' do
+        expect(subject.with_gov_delivery_error).not_to include(subscriber_list)
+      end
+    end
+
+    context 'when the topic does not exist' do
+      let(:fetch_topics_response) { nil }
```

Comment:
As above. Have added error catching around `fetch_topic` and `delete_topic` methods and appropriate testing